nmoscommon/utils.py

```diff
@@ -54,3 +77,82 @@ def getLocalIP():
                 pass
     # Could not find an interface
     return None
+
+
+def downgrade_api_version(
+    obj, rtype, target_ver, downgrade_ver=None, downgrade_map=DOWNGRADE_MAP, remove_keys=["max_api_version"]
+):
+
+    # Sort version list in ascending order
+    version_list = sorted(downgrade_map.keys())
+    # Add 'v1.0' as first element in version list
+    version_list.insert(0, 'v1.0')
+    # Set the max version as last element in list
+    max_ver = version_list[-1]
+
+    # Get current API version, or default to max API version
+    current_api_version = obj.get("@_apiversion", max_ver)
+
+    # Fail if target API version is greater than maximum
+    if api_ver_compare(target_ver, max_ver) > 0:
+        return None
+
+    # Downgrade API object, for a given resource, until it reaches target_version
+    while api_ver_compare(target_ver, current_api_version) < 0:
+        print("Processing downgrading from {}".format(current_api_version))
```

Comment:
Want me to get rid of this print @andrewbonney ? Might get a bit spammy

Reply:
Yeah, prints won't show up in our logs anyway. If anything it should use the logger with a debug level.